tests/models/test_marrmotm01.py

```diff
@@ -101,9 +94,76 @@ def test_get_value_as_xarray(self, model_with_setup):
         )
         assert_allclose(actual, expected)
 
-    def test_setup_with_own_work_dir(self, tmp_path, mocked_config, model: MarrmotM01, forcing_file: Path):
+    def test_setup_with_own_work_dir(self, tmp_path, mocked_config, model: MarrmotM01):
         cfg_file, cfg_dir = model.setup(
-            forcing=forcing_file,
             work_dir=tmp_path
         )
         assert cfg_dir == tmp_path
+
+
+class TestWithCustomSetupAndExampleData:
+    @pytest.fixture
+    def forcing_file(self):
+        # Downloaded from
+        # https://github.com/wknoben/MARRMoT/blob/master/BMI/Config/BMI_testcase_m01_BuffaloRiver_TN_USA.mat
+        return Path(__file__).parent / 'data' / 'BMI_testcase_m01_BuffaloRiver_TN_USA.mat'
+
+    @pytest.fixture
+    def model(self, forcing_file: Path, mocked_config):
+        m = MarrmotM01(version="2020.11", forcing=forcing_file)
+        yield m
+        if m.bmi:
+            # Clean up container
+            del m.bmi
+
+    @pytest.fixture
+    def model_with_setup(self, model: MarrmotM01):
+        cfg_file, cfg_dir = model.setup(
+            maximum_soil_moisture_storage=1234,
+            initial_soil_moisture_storage=4321,
+            start_time=datetime(1990, 1, 1, tzinfo=timezone.utc),
+            end_time=datetime(1991, 12, 31, tzinfo=timezone.utc),
+        )
+        return model, cfg_file, cfg_dir
+
+    def test_setup(self, model_with_setup):
```

Comment:
So this `model_with_setup` refers to `self.model_with_setup`?

Reply:
correct, model_with_setup is a fixture scoped to the class it is in